src/App.tsx

```diff
@@ -14,11 +16,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  start: boolean;
+  reversed: boolean;
+  sortBy: string;
+};
 
-export default App;
+enum SortBy {
+  Length = 'length',
+  Name = 'name',
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    start: false,
+    reversed: false,
+    sortBy: '',
+  };
+
+  start = () => {
+    this.setState({ start: true });
+  };
+
+  reverse = () => {
+    this.setState(state => ({ reversed: !state.reversed }));
+  };
+
+  reset = () => {
+    this.setState({
+      reversed: false,
+      sortBy: '',
+    });
+  };
+
+  sortByMethod = (sortArgument: SortBy): void => {
+    if (sortArgument === SortBy.Length) {
+      this.setState({ sortBy: SortBy.Length });
+    } else {
+      this.setState({ sortBy: SortBy.Name });
+    }
+  };
+
+  visibleGoods = () => {
```

Comment:
```suggestion
  getVisibleGoods = () => {
```
better name this method more meaningful, because now it sounds like a simple variable